Added function `preprocess_data()` to be imported from `ecog.signal_processing`

luiztauffer opened the pull request:
- Preprocess (resamples, filters) the raw data provided in a NWB file
- Runs spectral decomposition for bands in utils/bands.py
- Stores the processed signals and spectral decomposition power values in the same NWB file

Commit:
use preprocess_data() as an imported module

Commit:
add function preprocess_data to the signal_processing module

Commit:
Scales signals back to Volts after processing

bendichter commented:
it looks like cfs and sds are not input args

bendichter commented on ecog/signal_processing/preprocess_data.py:
```diff
@@ -0,0 +1,155 @@
+    block_path
+    rate
+    cfs: filter center frequencies. If None, use Chang lab defaults
+    sds: filer standard deviations. If None, use Chang lab defaults
```
these values are not in the function def

luiztauffer replied:
Oh I didn't look at the description, will do it now

bendichter commented on ecog/signal_processing/preprocess_data.py:
```diff
@@ -0,0 +1,155 @@
+        transform(block_path, phase=phase)
+
+
+def transform(block_path, suffix=None, phase=False, seed=20180928):
```
what is suffix?

luiztauffer replied:
it was present in the previous file, it was added at the end of the created file, if the user input something there. I'll remove it

bendichter reviewed (changes requested):
A few minor corrections. Overall it's looking really good!

Commit:
transform() function docstring.

JesseLivezey commented:
@luiztauffer @bendichter FYI, we're not planning on maintaining anything in this repo. It's likely that in a few weeks we'll make the current version of our preprocessing code public, which was initially based on the code here. Were you looking to build substantially on this code?

luiztauffer commented:
Hi @JesseLivezey   thanks for the heads up!
We’re altering the current process_ecog for two main reasons:
1 – make the preprocessing function directly usable by an [ECoG visualization](https://github.com/bendichter/ecogVIS) project.
2 – make it store the results in NWB files

Would you mind sharing what will be the main changes with the new software? =)

Commit:
preprocess_data update
1 – user defined filtering bands
2 – docstring improved
3 – LFP and 3 decomposition types (default, high-gamma, custom) can be 
stored in NWB file
4 – checks if containers already exist and create only new ones

Commit:
minor correction on docstring

Commit:
remove hamming window filtering
High gamma band can be later calculated using the average power of 
selected gaussian filtered bands

JesseLivezey commented:
It might be easier to just copy the preprocessing functions and modify them in your repo. I could also make a different branch and merge them here (I'm using the docker branch for something specific).

The low-level changes to the new version are mostly for operating on arrays that are too large to process in memory all at once. Otherwise, they are the same. The high-level changes are to make the later stages of processing NWB compatible (and also somewhat specific to our lab).

JesseLivezey commented:
I can't commit to managing anything in this repo, so it might be better for the other project not to depend on anything here.
